Micronaut 3 on GraalVM Native Image not working on JDK 21

### Expected Behavior

Expect Micronaut on GraalVM Native Image to work on JDK 21.

### Actual Behaviour

Exception thrown:
```
java.lang.RuntimeException: java.lang.NoSuchMethodException: java.nio.DirectByteBuffer.<init>(long,int)
	at io.micronaut.buffer.netty.NettyFeature.beforeAnalysis(NettyFeature.java:70)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$9(NativeImageGenerator.java:765)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:86)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:765)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:585)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:542)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:407)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:611)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:133)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:93)
Caused by: java.lang.NoSuchMethodException: java.nio.DirectByteBuffer.<init>(long,int)
	at java.base/java.lang.Class.getConstructor0(Class.java:3689)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2858)
	at io.micronaut.buffer.netty.NettyFeature.beforeAnalysis(NettyFeature.java:68)
	... 9 more
```

This is due to this private constructor was removed in JDK 21: https://bugs.openjdk.org/browse/JDK-8303083

On the 3.10.x branch, the constructor is still referenced: https://github.com/micronaut-projects/micronaut-core/blob/3.10.x/buffer-netty/src/main/java/io/micronaut/buffer/netty/NettyFeature.java#L66

### Steps To Reproduce

Get an early access JDK 21, e.g. from here: https://jdk.java.net/21/

Build native image from tip (to get preliminary JDK 21 support):
```
git clone https://github.com/oracle/graal.git
cd graal/substratevm
mx --java-home path/to/jdk21 build
export GRAALVM_HOME=$(mx --java-home path/to/jdk21 graalvm-home)
```

Download the micronaut + native image example project from https://guides.micronaut.io/latest/micronaut-creating-first-graal-app-gradle-java.html

```
cd micronaut-creating-first-graal-app-gradle-java 
./gradlew nativeCompile info
...
========================================================================================================================
GraalVM Native Image: Generating 'micronautguide' (executable)...
========================================================================================================================
[1/8] Initializing...                                                                                    (3.1s @ 0.16GB)
 Java version: 21-ea+19-LTS-1566, vendor version: Oracle GraalVM 21-dev.ea+19.1
 Graal compiler: optimization level: '2', target machine: 'x86-64-v3'
 C compiler: gcc (redhat, x86_64, 12.2.1)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 3 user-specific feature(s)
 - io.micronaut.buffer.netty.NettyFeature
 - io.micronaut.core.graal.ServiceLoaderFeature
 - io.micronaut.http.netty.graal.HttpNettyFeature

------------------------------------------------------------------------------------------------------------------------
                         0.2s (4.4% of total time) in 9 GCs | Peak RSS: 0.81GB | CPU load: 6.57
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /home/zapster/tmp/micronaut-creating-first-graal-app-gradle-java/build/native/nativeCompile/svm_err_b_20230509T175344.119_pid585855.md (build_info)
========================================================================================================================
Failed generating 'micronautguide' after 3.2s.

The build process encountered an unexpected error:

> java.lang.RuntimeException: java.lang.NoSuchMethodException: java.nio.DirectByteBuffer.<init>(long,int)

Please inspect the generated error report at:
/home/zapster/tmp/micronaut-creating-first-graal-app-gradle-java/build/native/nativeCompile/svm_err_b_20230509T175344.119_pid585855.md

```


### Environment Information

- linux-amd64
- JDK 21 EA

### Example Application

https://guides.micronaut.io/latest/micronaut-creating-first-graal-app-gradle-java.html

### Version

3.10.x

--- a/buffer-netty/src/main/java/io/micronaut/buffer/netty/NettyFeature.java
+++ b/buffer-netty/src/main/java/io/micronaut/buffer/netty/NettyFeature.java
@@ -64,8 +64,8 @@ final class NettyFeature implements Feature {
         ImageSingletons.lookup(SystemPropertiesSupport.class).initializeProperty("io.netty.tryReflectionSetAccessible", "true");
         try {
             RuntimeReflection.register(access.findClassByName("java.nio.DirectByteBuffer").getDeclaredConstructor(long.class, int.class));
-        } catch (NoSuchMethodException e) {
-            throw new RuntimeException(e);
+        } catch (NoSuchMethodException ignored) {
+            // not present on JDK 21
         }
         Class<?> unsafeOld = access.findClassByName("sun.misc.Unsafe");
         if (unsafeOld != null) {

